File: youlexuan-user-interface/src/main/java/com/offcn/user/service/UserService.java
package com.offcn.user.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbUser;

import java.util.List;

/*
用户服务层接口
 */
public interface UserService {

    /*
    全部用户列表
     */
    public List<TbUser> selectAll();

    /*
    分页列表
     */
    public PageResult selectPage(int pageNum, int pageSize);

    /*
    增加用户
     */
    public void insert(TbUser user);

    /*
    修改用户
     */
    public void update(TbUser user);

    /*
    根据ID获取实体
     */
    public TbUser selectOne(Long id);

    /*
    删除用户
     */
    public void delete(Long[] ids);

    /*
    条件查询
     */
    public PageResult findPage(TbUser user, int pageNum, int pageSize);
}

File: youlexuan-manager-web/src/main/webapp/js/service/userService.js
/*
*定义服务
*/
app.service('userService', function ($http) {

    //获取所有的品牌信息
    this.selectAll = function () {
        return $http.get('../user/selectAll.do');
    };

    //分页
    this.selectPage = function (page, rows) {
        return $http.get('../user/selectPage.do?page=' + page + '&rows=' + rows);
    };

    //增加品牌
    this.insert = function (entity) {
        return $http.post("../user/insert.do", entity);
    };

    //更新品牌
    this.update = function (entity) {
        return $http.post("../user/update.do", entity);
    };

    //查询单个
    this.selectOne = function (id) {
        return $http.get("../user/selectOne.do?id=" + id);
    };

    //删除品牌
    this.delete = function (ids) {
        return $http.get("../user/delete.do?ids=" + ids);
    };

    //条件查询
    this.search = function (page, rows, searchEntity) {
        return $http.post('../user/search.do?page=' + page + "&rows=" + rows, searchEntity);
    };

})